### PR TITLE
Adding ref doc for cadl compiler

### PR DESCRIPTION
**Pull request by allenjzhang**

### CODE REVIEW

File: packages/website/docusaurus.config.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const config = {
   tagline: "API first with Cadl for Azure services",
   url: "https://microsoft.github.io",
   baseUrl: process.env.CADL_WEBSITE_BASE_PATH ?? "/",
-  onBrokenLinks: "throw",
+  onBrokenLinks: "warn",
```
Comment (allenjzhang) [inline packages/website/docusaurus.config.js:43]: Looks like there is a bug in Docusarus on in-page MD link. Set to warn.

File: packages/website/docusaurus.config.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ const config = {
   tagline: "API first with Cadl for Azure services",
   url: "https://microsoft.github.io",
   baseUrl: process.env.CADL_WEBSITE_BASE_PATH ?? "/",
-  onBrokenLinks: "throw",
+  onBrokenLinks: "log",
```
Comment (nguerrera) [inline packages/website/docusaurus.config.js:43]: Do we want to keep this?

**Review by nguerrera — APPROVED** [review]

File: packages/website/docusaurus.config.js (hunk 2 above)
Comment (allenjzhang) [reply packages/website/docusaurus.config.js:43]: I suspect bug in docusaurus. The reported link in md was correctly set. Opened an investigation bug to track this. #1574

Comment (nguerrera) [reply packages/website/docusaurus.config.js:43]: Sounds good, thanks.

**Pull request merged**